compiler/src/dotty/tools/dotc/semanticdb/TypeOps.scala

```diff
@@ -228,6 +233,14 @@ class TypeOps:
           val ssym = sym.symbolName
           s.TypeRef(spre, ssym, Seq.empty)
 
+        case TypeRef(pre, sym: Name) =>
+          // val spre = if tpe.hasTrivialPrefix then s.Type.Empty else loop(pre)
+          // Craft semanticdb symbol
+          // prefix symbol (e.g. "scala/") + name (e.g. "Nothing") + Global symbol suffix ("#")
+          // see: https://scalameta.org/docs/semanticdb/specification.html#symbol
+          val ssym = s"${pre.typeSymbol.symbolName}${sym.mangledString}#"
+          s.TypeRef(s.Type.Empty, ssym, Seq.empty)
```

Comment:
`x *: xs` part of
```scala
type Concat[Xs <: Tuple, +Ys <: Tuple] <: Tuple = Xs match
  case EmptyTuple => Ys
  case x *: xs => x *: Concat[xs, Ys]
```

have `TypeRef(..., Nothing)` (where `Nothing` is just a `Name` instead of `Symbol`), and we can't access the symbol for `Nothing` here. Therefore, craft the symbol for `TyperRef` for this.